docs/1.getting-started/7.state-management.md

```diff
@@ -57,9 +57,68 @@ const counter = useState('counter', () => Math.round(Math.random() * 1000))
 To globally invalidate cached state, see [`clearNuxtState`](/docs/api/utils/clear-nuxt-state) util.
 ::
 
-### Advanced Usage
+### Initializing State
 
-In this example, we use a composable that detects the user's default locale from the HTTP request headers and keeps it in a `locale` state.
+Most of the time you will want to initialize your state with async data, you can use the [`app.vue`](/docs/guide/directory-structure/app) component to do so with the [`once`](/docs/api/utils/once) util.
+
+```vue [app.vue]
+<script setup>
+const websiteConfig = useState('config')
+
+await once(async () => {
+  websiteConfig.value = await $fetch('https://my-cms.com/api/website-config')
+})
+</script>
+```
+
+::callout
+This is similar to Nuxt 2 `nuxtServerInit` action to fill the store with data before rendering the page.
```

Comment:
Would be nice to add a link for anyone unfamiliar with Nuxt 2 :) Reworded it slightly, feel free to modify even further if needed!
```suggestion
This is similar to the [`nuxtServerInit` action](https://v2.nuxt.com/docs/directory-structure/store/#the-nuxtserverinit-action) in Nuxt 2, which allows filling the initial state of your store server-side before rendering the page on the client side.
```